src/lib/OS/SyscallHandler.cc

```diff
@@ -1,15 +1,756 @@
 #include "simeng/OS/SyscallHandler.hh"
 
 #include "simeng/OS/SimOS.hh"
+
 namespace simeng {
 namespace OS {
 
-SyscallHandler::SyscallHandler(SimOS* os) : os_(os) {
+SyscallHandler::SyscallHandler(SimOS* OS,
+                               std::shared_ptr<simeng::memory::Mem> memory)
+    : OS_(OS), memory_(memory) {
   // Define vector of all currently supported special file paths & files.
   supportedSpecialFiles_.insert(
       supportedSpecialFiles_.end(),
       {"/proc/cpuinfo", "proc/stat", "/sys/devices/system/cpu",
        "/sys/devices/system/cpu/online", "core_id", "physical_package_id"});
+
+  resumeHandling_ = [this]() { handleSyscall(); };
```

Comment:
Is `resumeHandling_` needed anymore, given that there are only two places `resumeHandling` is assigned a new value i.e  the constructor and `concludeSyscall`, and both time the value assigned is the same.

Could not just do:
```
SycallHandler.tick() {
if (!syscallqueue_.empty()) {
handleSyscall()
}
}
```

Reply:
I think the resumeHandling setup is a good way to deal with non-instant memory accesses within the handler. Something that will be needed with SST but also not impossible for native SimEng

Reply:
Immediately it doesn't add any benefit right? I assume for SST we will have a  class derived from SimOS, which could then be handled separately given that we have now made SimOS the dependency of SyscallHandler again. So wouldn't it mean that this logic will remain the same?

Reply:
It is just confusing to understand, that is all. If you don't think it should be changed then please add a comment as to why the logic exists in such a way

Reply:
Not sure about the derived from SimOS comment, the syscallHandler will be the same for both native SimEng and SST.

Reply:
ok resolving it for now, this should be discussed in the future though

Reply:
I'll remove it as it doesn't do anything at the moment and we'll deal with multi-cycle responses from memory when we get to it